src/thunks/searchThunk.ts

```diff
@@ -1,74 +1,99 @@
-    );
-  } catch (err) {
-    return rejectWithValue((err as Error).message);
-  }
-  const json = (await response.json()) as APIPushResponse;
-  if (json.results.length > 0) {
-    return json.results;
-  }
+  RevisionsList[],
+  FetchDataArgs,
+  { rejectValue: FetchDataMeta }
+>(
+  'search/fetchRecentRevisions',
+  async ({ repository, searchType }, { rejectWithValue }) => {
+    let response;
+
+    try {
+      response = await fetch(
+        `${treeherderBaseURL}/api/project/${repository}/push/?hide_reviewbot_pushes=true`,
+      );
+    } catch (err) {
+      const error = (err as Error).message;
+      return rejectWithValue({ error, searchType });
+    }
+    const json = (await response.json()) as APIPushResponse;
+    if (json.results.length > 0) {
+      return json.results;
+    }
 
-  return rejectWithValue('No results found');
-});
+    return rejectWithValue({ error: 'No results found', searchType });
+  },
+);
 
 export const fetchRevisionByID = createAsyncThunk<
-  Revision[],
-  { repository: Repository['name']; search: string },
-  { rejectValue: string }
+  RevisionsList[],
+  FetchDataArgsIDEmail,
+  { rejectValue: FetchDataMeta }
 >(
   'search/fetchRevisionByID',
-  async ({ repository, search }, { rejectWithValue }) => {
+  async ({ repository, search, searchType }, { rejectWithValue }) => {
     let response;
     try {
       response = await fetch(
         `${treeherderBaseURL}/api/project/${repository}/push/?revision=${search}`,
       );
     } catch (err) {
-      return rejectWithValue((err as Error).message);
+      const error = (err as Error).message;
+      return rejectWithValue({ error, searchType });
     }
     const json = (await response.json()) as APIPushResponse;
     if (json.results.length > 0) {
       return json.results;
     }
 
-    return rejectWithValue('No results found');
+    return rejectWithValue({ error: 'No results found', searchType });
   },
 );
 
 export const fetchRevisionsByAuthor = createAsyncThunk<
-  Revision[],
-  { repository: Repository['name']; search: string },
-  { rejectValue: string }
+  RevisionsList[],
+  FetchDataArgsIDEmail,
+  { rejectValue: FetchDataMeta }
 >(
   'search/fetchRevisionsByAuthor',
-  async ({ repository, search }, { rejectWithValue }) => {
+  async ({ repository, search, searchType }, { rejectWithValue }) => {
     let response;
     try {
       response = await fetch(
         `${treeherderBaseURL}/api/project/${repository}/push/?author=${search}`,
       );
     } catch (err) {
-      return rejectWithValue((err as Error).message);
+      const error = (err as Error).message;
+      return rejectWithValue({ error, searchType });
     }
     const json = (await response.json()) as APIPushResponse;
     if (json.results.length > 0) {
       return json.results;
     }
 
-    return rejectWithValue('No results found');
+    return rejectWithValue({ error: 'No results found', searchType });
```

Comment:
In the same logic as above, it would be better to reject with a full Error object (`new Error("No results found")`) so that there's a stack. This could be handled in a follow-up.